src/Datadog.Trace/Logging/LogRateLimiter.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections.Concurrent;
+
+using Datadog.Trace.Util;
+
+namespace Datadog.Trace.Logging
+{
+    internal class LogRateLimiter : ILogRateLimiter
+    {
+#if NET45
+        private static readonly DateTime _unixEpoch = new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc);
+#endif
+
+        private readonly int _secondsBetweenLogs;
+        private readonly ConcurrentDictionary<int, LogRateBucketInfo> _buckets
+            = new ConcurrentDictionary<int, LogRateBucketInfo>();
+
+        public LogRateLimiter(int secondsBetweenLogs)
+        {
+            if (secondsBetweenLogs < 1)
+            {
+                throw new ArgumentException("Must have positive number of seconds between logs", nameof(secondsBetweenLogs));
+            }
+
+            _secondsBetweenLogs = secondsBetweenLogs;
+        }
+
+        /// <inheritdoc/>
+        public bool ShouldLog(string filePath, int lineNumber, out uint skipCount)
```

Comment:
I don't know if it may actually happen, but maybe we should have a special path for `filePath == string.Empty && lineNumber == 0` where we don't filter the logs, just in case the CallerLineNumber/CallerFilePath attributes start misbehaving for a weird reason

Reply:
Yeah, good idea. _Shouldn't_ be a reason for that, but would be v. harmful if it did!

Reply:
Added in https://github.com/DataDog/dd-trace-dotnet/pull/1153/commits/05de3984613bd001f05f2c5f22b34db3ccb4d2e5